app/coffee/Features/FileStore/FileStoreHandler.coffee

```diff
@@ -25,10 +25,19 @@ module.exports = FileStoreHandler =
 				timeout:fiveMinsInMs
 			writeStream = request(opts)
 			readStream.pipe writeStream
-			writeStream.on "end", callback
+
+			writeStream.on 'response', (response) ->
+				if response.statusCode not in [200, 201]
```

Comment:
I often use the pattern 200 <= statusCode < 300 to catch any 'success' code, but I think this is fine here.